### ISSUE TITLE
[bug] Orphaned process "node -r esbuild-register .../loadFile.js" after closing dev server

### ISSUE THREAD

**Issue by nderscore** [tamagui/tamagui#492]
Seen on: v1.0.2

**Describe the bug**

After running Next.js or Expo dev servers with Tamagui plugins enabled, an orphaned node process is left behind with the following arguments:

```bash
node -r esbuild-register /path/to/my/project/node_modules/@tamagui/static/dist/extractor/loadFile.js
```

**To Reproduce**

Using the untouched starter project, launch Next or Expo dev server, then after it has loaded, send a sigint (ctrl+c)

Check running processes:

```bash
ps aux | grep loadFile
```

**System Info:** 

OS: macOS
CPU: arm64 (M1/M2)
Node: Replicated on both v16.18.1 and v18.12.1
Yarn: v3.x

**Investigation**

It looks like the child process should be killed when the parent terminates:

https://github.com/tamagui/tamagui/blob/c6d31da34ac2ab2a49545a21f0a395cf0564bb38/packages/static/src/extractor/getStaticBindingsForScope.ts#L40-L51

This callback function executes successfully (`child.kill()` returns `true`), and even kills the process.

But, immediately afterwards, the script for `getStaticBindingsForScope` is re-evaluated and spawns a new forked process 🤔

**Comment by nderscore** [comment]
Deferring the creation of the child process until first use seems to fix the issue for me...

```diff
diff --git a/packages/static/src/extractor/getStaticBindingsForScope.ts b/packages/static/src/extractor/getStaticBindingsForScope.ts
index e39ef29b01..9b4209a839 100644
--- a/packages/static/src/extractor/getStaticBindingsForScope.ts
+++ b/packages/static/src/extractor/getStaticBindingsForScope.ts
@@ -1,4 +1,4 @@
-import { fork } from 'child_process'
+import { ChildProcess, fork } from 'child_process'
 import { dirname, extname, join, resolve } from 'path'
 
 import { Binding, NodePath } from '@babel/traverse'
@@ -31,13 +31,17 @@ setInterval(() => {
 const loadCmd = `${join(__dirname, 'loadFile.js')}`
 
 let exited = false
-const child = fork(loadCmd, [], {
-  execArgv: ['-r', 'esbuild-register'],
-  detached: false,
-  stdio: 'ignore',
-})
+let child: ChildProcess
+function forkChild() {
+  child = fork(loadCmd, [], {
+    execArgv: ['-r', 'esbuild-register'],
+    detached: false,
+    stdio: 'ignore',
+  })
+}
 
 export function cleanupBeforeExit() {
+  if (!child) return
   if (exited) return
   child.removeAllListeners()
   child.unref()
@@ -51,6 +55,9 @@ process.once('SIGINT', cleanupBeforeExit)
 process.once('beforeExit', cleanupBeforeExit)
 
 function importModule(path: string) {
+  if (!child) {
+    forkChild()
+  }
   if (pending.has(path)) {
     return pending.get(path)
   }
```

But I'm still not really sure _why_ the module-scope code in this file re-executes during termination in the first place 🤔

**Comment by natew** [comment]
Good point, can do a more thorough fix later but for now fixed in v1.0.3 04a9732d00ab35c9d8a04d64ccb5b0362be0e5e1
